src/service.rs

```diff
@@ -810,57 +994,81 @@ impl Collection {
         T::from_msgpack(&decrypted)
     }
 
+    /// Set metadata for the collection object from a byte array
+    /// 
+    /// # Arguments:
+    /// * `meta` - the metadata for the collection. This needs to be a valid [ItemMetadata] struct encoded using `msgpack`.    
     pub fn set_meta_raw(&mut self, meta: &[u8]) -> Result<()> {
         self.col.set_meta(&self.cm, &meta)
     }
 
+    /// Return metadata for the collection object as a byte array
     pub fn meta_raw(&self) -> Result<Vec<u8>> {
         self.col.meta(&self.cm)
     }
 
+    /// Set the content of the collection
+    /// 
+    /// # Arguments:
+    /// * `content` - the content of the collection as a byte array 
     pub fn set_content(&mut self, content: &[u8]) -> Result<()> {
         self.col.set_content(&self.cm, content)
     }
 
+    /// Return the content of the collection as a byte array
     pub fn content(&self) -> Result<Vec<u8>> {
         self.col.content(&self.cm)
     }
 
+    /// Mark the collection as deleted
```

Comment:
Maybe in the long description (for this, and for the one for item) mention that the item is only marked for deletion and still needs to be uploaded?